doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,79 @@
+# Architecture
+
+*(This chapter is unstable and a work in progress. Incoming links may rot.)*
+
+This chapter describes how Nix works.
```

Comment:
This may sound like splitting hairs, but instead of the word "Nix", the manuals should say explicitly what the term refers to, even if it is to the detriment of brevity.<sup><b>1</b></sup> Depending on the context, I saw it used to mean:

+ Nix core concepts / paradigm / model
+ Nix CLI tools
+ Nix expression language
+ the implementation of Nix (i.e., the `NixOS/nix`)
+ the Nix ecosystem as a whole (i.e., `NixOS/nix` implementation, Nixpkgs, NixOS, even NixOps)
+ Nixpkgs (rarely, but it still)
+ (there is probably more..)

Taking the very next sentence as an example:
> It should help users understand why Nix behaves as it does, and it should help developers understand how to modify Nix and how to write similar tools.

+ "_NIx behaves_": I presume this alludes to the Nix model and/or the implementation
+ "_to modify Nix_": To modify a Nix expression, the behavior of the Nix CLI tools, the store object in the Nix store - or something else?

I guess the "Architecture" section is about the model itself, and if so, it would be prudent to start with a short explanation what it is about. (It would be nice to have a specification.) Below are two ways I tried to define "Nix model" to myself:

1. _A purely functional deployment model_
  ... then explain what purity<sup><b>2</b></sup>, "functional" ([this](https://github.com/NixOS/nix/blob/006e8f5b450c9524fe774b35954d342ee1c50d50/doc/manual/src/architecture/store/abstract/abstract.md) from PR #6877), and [deployment model](https://stackoverflow.com/a/73229950/1498178) mean.
  (Took it directly from the thesis; the omission of the word "software" is deliberate.)

2. _A (new) paradigm of building anything that requires assembly<sup><b>3</b></sup>, and managing the build results. The official term for "build result" is **store object** because they are kept in the Nix store._

---

\[1]: The definition of "Nix" has been a subject of contention on many online forums, but personally, it even made stop trying out Nix in the beginning and to give up on more certain parts of it a couple of times.  I found the tutorials, posts, manuals, etc. confusing because sometimes there were sudden context switches, and not knowing what other aspects of NIx there are, I couldn't connect the dots (heh, still can't most of the time).

\[2]: [Thesis](https://edolstra.github.io/pubs/phd-thesis.pdf), page 21 (PDF page 29), 2nd paragraph from bottom.

\[3]: My first try was "_A (new) paradigm of building digital artifacts<sup><b>3</b></sup>  (where the input can be anything that require assembly), and managing the build results._, but the term "digital artifact" seems just as overloaded as "component" or "package":
+ Model-Based Systems Engineering (MBSE) wiki: [Digital Artifact](https://www.omgwiki.org/MBSE/doku.php?id=mbse:digital_artifact)
+ Wikipedia: [Digital artifact](https://en.wikipedia.org/wiki/Digital_artifact)
+ WikiEducator: [Digital artefact](https://wikieducator.org/Digital_artefact)
+ Quora: [What is meant by digital artifact?](https://www.quora.com/What-is-meant-by-digital-artifact)